trino/exceptions.py

```diff
@@ -72,7 +72,7 @@ class TrinoDataError(NotSupportedError):
 
 
 class TrinoQueryError(Error):
-    def __init__(self, error: Dict[str, Any], query_id: Optional[str] = None) -> None:
+    def __init__(self, error: Any, query_id: Optional[str] = None) -> None:
```

Comment:
Doesn't the error come from JSON, so I would expect it to follow some certain format?

Reply:
I would expect the same, but [here](https://github.com/trinodb/trino-python-client/blob/57c97061ac1735675d4cf43da16b550dcb15578e/trino/client.py#L858) `str` is passed as an argument to `error` parameter.
It is causing mypy error:
```
trino/client.py:858: error: Argument 1 to "TrinoUserError" has incompatible type "str"; expected "Dict[str, Any]"  [arg-type]
```

Reply:
I would say that we need to fix that occurrence. If you have a look at `TrinoQueryError`, you'll see this is actually an issue.